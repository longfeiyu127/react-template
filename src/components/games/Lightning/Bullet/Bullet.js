import React from 'react';
// import Tictactoe from './tictactoe'
// import Fivechess from './fivechess'
import './Bullet.less';

class Bullet extends React.Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    // eslint-disable-next-line
    const { position, bulletTpey, size } = this.props;
    return (
      <div
        className="g-lig-bullet"
        style={{
          top: `${position.top}rem`,
          left: `${position.left}rem`,
          height: `${size.height}rem`,
          width: `${size.width}rem`
        }}
      >
        {/* style={{fontWeight: (move === stepNumber) ? 900 : 500, cursor: 'pointer'}} */}
        <main className={['bullet-main', bulletTpey].join(' ')} />
      </div>
    );
  }
}

export default Bullet;
