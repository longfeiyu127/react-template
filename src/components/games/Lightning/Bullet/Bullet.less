.g-lig-bullet{
  position: absolute;
  >.bullet-main {
    height: 100%;
    width: 100%;
  }
  >.player{
    background: url('../../../../assets/images/games/lightning/bullet/palyerBullet.png') no-repeat center/100% 100%;
    animation: flicker 100ms infinite;
  }

  @keyframes flicker {
    0%, 100% {
      opacity: 0;
    }
    50% {
      opacity: 1;
    }
  }
}
