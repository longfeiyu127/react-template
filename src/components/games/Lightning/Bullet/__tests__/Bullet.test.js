import React from 'react';
import renderer from 'react-test-renderer';
import Bullet from '../Bullet';

describe('<Bullet />', () => {
  it('component render', () => {
    const position = {
      top: 0.1,
      left: 0.1
    };
    const size = {
      height: 0.1,
      width: 0.1
    };
    let bullet = renderer
      .create(<Bullet position={position} size={size} bulletTpey="player" />)
      .toJSON();
    expect(bullet).toMatchSnapshot();

    bullet = renderer
      .create(<Bullet position={position} size={size} bulletTpey="player" />)
      .toJSON();
    expect(bullet).toMatchSnapshot();
  });
});
