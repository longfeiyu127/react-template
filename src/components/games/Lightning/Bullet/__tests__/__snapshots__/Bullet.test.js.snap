// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bullet /> component render 1`] = `
<div
  className="g-lig-bullet"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="bullet-main player"
  />
</div>
`;

exports[`<Bullet /> component render 2`] = `
<div
  className="g-lig-bullet"
  style={
    Object {
      "height": "0.1rem",
      "left": "0.1rem",
      "top": "0.1rem",
      "width": "0.1rem",
    }
  }
>
  <main
    className="bullet-main player"
  />
</div>
`;
